### PR TITLE
buildGoPackage: check/delete vendor

### PR DESCRIPTION
**Pull request by zowoq**
I've seen several existing and newly submitted packages using an unnecessary `goDeps` with `buildGoPackage`, hopefully this would point people in the right direction.

Copied `deleteVendor` from `buildGoModule` to cover cases when we don't want to use the vendor folder.

~~Would need to rebuild/fix packages and update docs before merging.~~ Done.

cc @NixOS/golang

### CODE REVIEW

**Comment by zowoq** [comment]
Need to fix a couple of packages that are half vendored.

File: pkgs/servers/livepeer/default.nix
Comment (zowoq) [inline pkgs/servers/livepeer/default.nix:19]: Out of ~390 `buildGoPackage`s this seems to be the only one that breaks and it's non-trivial to fix.

I'll come back to it after this PR is merged.

**Review by mdlayher — APPROVED** [review]
LGTM, nice work.

**Review by kalbasit — APPROVED** [review]

**Comment by zowoq** [comment]
I'll merge after the branch off.

**Comment by zowoq** [comment]
Rechecked `nixpkgs-review` on nixos x86_64.

**Pull request merged**
